tModBuildTasks/SynchronizeDirectories.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+
+namespace tModLoader.BuildTasks;
+
+/// <summary>
+/// Copies contents from one folder to another, additionally removing entries in the destination that aren't present in the source.
+/// </summary>
+public class SynchronizeDirectories : TaskBase
+{
+	[Required]
+	public string Source { get; set; } = string.Empty;
+
+	[Required]
+	public string Destination { get; set; } = string.Empty;
+
+	protected override void Run()
+	{
+		Source = Path.GetFullPath(Source);
+		Destination = Path.GetFullPath(Destination);
+
+		var source = new DirectoryInfo(Source);
+		var destination = new DirectoryInfo(Destination);
+
+		if (!source.Exists) {
+			Log.LogError($"Source directory '{Source}' doesn't exist!");
+			return;
+		}
+
+		destination.Create();
+
+		// Delete files that shouldn't be there.
+		Parallel.ForEach(destination.EnumerateFiles("*", SearchOption.AllDirectories), file => {
+			string relativePath = IOUtils.SubstringToRelativePath(file.FullName, Destination);
+			string sourcePath = Path.Combine(Source, relativePath);
+
+			if (!File.Exists(sourcePath)) {
+				IOActionWrapper(file.Delete);
```

Comment:
Does this need `FileNotFoundException` checking? Like the `DirectoryNotFoundException` one below? I assume it's a race condition fix.

Reply:
Would only prevent user-caused races in case of files, but doesn't hurt to have.